sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -209,20 +479,291 @@ node add_malloc_device(void *&data, size_t numBytes, const std::vector<node>& de
 ----
 |Adding a node that encapsulates a `malloc` operation.
 
+Parameters:
+
+* `data` - Return parameter set to the address of memory allocated.
+
+* `numBytes` - Size in bytes to allocate.
+
+* `dep` - Nodes the created node will be dependent on.
+
+Returns: The memory allocation node which has been added to the graph
+
 |
 [source,c++]
 ----
 node add_free(void *data, const std::vector<node>& dep = {});
 ----
 |Adding a node that encapsulates a `free` operation.
 
+Parameters:
+
+* `data` - Address of memory to free.
+
+* `dep` - Nodes the created node will be dependent on.
+
+Returns: The memory freeing node which has been added to the graph.
+
+Exceptions:
+
+* TODO - Throw if not allocated by `add_malloc_device`?
+* TODO - Throw if already freed?
+* TODO - Throw if not valid address?
```

Comment:
Should `add_free` error under any of these circumstances?

Reply:
Tracked in https://github.com/reble/llvm/issues/10, and will remove the TODOs for this PR